src/components/AddPaymentMethodMenu.js

```diff
@@ -37,42 +47,63 @@ const propTypes = {
     /** Popover anchor ref */
     anchorRef: refPropTypes,
 
-    ...withLocalizePropTypes,
+    /** Session info for the currently logged in user. */
+    session: PropTypes.shape({
+        /** Currently logged in user accountID */
+        accountID: PropTypes.number,
+    }),
 };
 
 const defaultProps = {
+    iouReport: {},
     anchorPosition: {},
     anchorAlignment: {
         horizontal: CONST.MODAL.ANCHOR_ORIGIN_HORIZONTAL.LEFT,
         vertical: CONST.MODAL.ANCHOR_ORIGIN_VERTICAL.BOTTOM,
     },
     betas: [],
     anchorRef: () => {},
+    session: {},
 };
 
-function AddPaymentMethodMenu(props) {
+function AddPaymentMethodMenu({isVisible, onClose, anchorPosition, anchorAlignment, anchorRef, iouReport, onItemSelected, session, betas}) {
+    const {translate} = useLocalize();
+
     return (
         <PopoverMenu
-            isVisible={props.isVisible}
-            onClose={props.onClose}
-            anchorPosition={props.anchorPosition}
-            anchorAlignment={props.anchorAlignment}
-            anchorRef={props.anchorRef}
-            onItemSelected={props.onClose}
+            isVisible={isVisible}
+            onClose={onClose}
+            anchorPosition={anchorPosition}
+            anchorAlignment={anchorAlignment}
+            anchorRef={anchorRef}
+            onItemSelected={onClose}
             menuItems={[
-                {
-                    text: props.translate('common.bankAccount'),
-                    icon: Expensicons.Bank,
-                    onSelected: () => {
-                        props.onItemSelected(CONST.PAYMENT_METHODS.BANK_ACCOUNT);
-                    },
-                },
-                ...(Permissions.canUseWallet(props.betas)
+                ...(ReportUtils.isIOUReport(iouReport)
                     ? [
                           {
-                              text: props.translate('common.debitCard'),
+                              text: translate('common.personalBankAccount'),
+                              icon: Expensicons.Bank,
+                              onSelected: () => {
+                                  onItemSelected(CONST.PAYMENT_METHODS.PERSONAL_BANK_ACCOUNT);
+                              },
+                          },
+                      ]
+                    : []),
+                ...(!ReportActionsUtils.hasRequestFromCurrentAccount(lodashGet(iouReport, 'reportID', 0), lodashGet(session, 'accountID', 0))
+                    ? [
+                          {
+                              text: translate('common.businessBankAccount'),
+                              icon: Expensicons.Building,
+                              onSelected: () => onItemSelected(CONST.PAYMENT_METHODS.BUSINESS_BANK_ACCOUNT),
```

Comment:
This caused crash - #31797 as it fallback to throw here:
https://github.com/Expensify/App/blob/7a68a3d1003130e63065bcbef253c912a3431c08/src/pages/settings/Wallet/WalletPage/WalletPage.js#L210